src/refreshtoken.ts

```diff
@@ -0,0 +1,100 @@
+import { SessionResponse, SessionResponseData, RefreshTokenRenewResponse } from './interfaces'
+
+const REFRESH_TOKEN_API = '/api/vtexid/refreshtoken/webstore'
+const LOCAL_STORAGE_KEY = 'vid_rt_webstore'
+const STATUS = {
+  SUCCESS: 'success',
+  INVALID_SESSION: 'invalidsession',
+  ERROR: 'error'
+}
+const TTL_30_MIN = 30 * 60 * 1000
+const TTL_12_HOURS = 12 * 60 * 60 * 1000
+const APPNAME = 'render-session'
+const TIMEOUT = 5000
+
+const setItem = (status: string, refreshAfter: Date) =>
+  localStorage.setItem(
+    LOCAL_STORAGE_KEY,
+    JSON.stringify({ status, refreshAfter })
+  )
+
+const removeKey = () => {
+  localStorage.removeItem(LOCAL_STORAGE_KEY)
+}
+
+const setError = () =>
+  setItem(STATUS.ERROR, new Date(new Date().getTime() + TTL_30_MIN))
+
+const fetchWithTimeout = (url: string, options: RequestInit | undefined, timeout: number): Promise<Response> => {
+  return new Promise((resolve, reject) => {
+    setTimeout(() => reject(new Error("timeout")), timeout)
+    fetch(url, options).then(resolve, reject)
+  })
+}
+
+export const tryInsertFirstRenewKey = (sessionResponse: SessionResponse) => {
+  try {
+    const localStorage = window.localStorage
+    if (!localStorage) return
+  
+    const isAuthenticated = (sessionResponse.response as SessionResponseData).namespaces.profile.isAuthenticated.value === "true"
```

Comment:
I know you are protected by a try/finally, but I prefer to always handle the known errors instead of letting it explode.

After opening https://storetheme.vtex.com with no cookies I noticed `profile` does not exist in `namespaces`, so the `profile.isAuthenticated` would throw a `Cannot read property 'isAuthenticated' of undefined`.

I suggest we deal with this case.

Reply:
Makes sense. Changed!